src/driver/communication_interface.cc

```diff
@@ -478,12 +502,25 @@ void CommunicationInterface::HandleSimDriverEventTrigger(
         "to simulate control exception!");
     sim_control_exception_triggered_ = true;
     return;
+  }
+  if (desired_event == "u_stop") {
+    if (robot_data_.robot_state.robot_mode != franka::RobotMode::kUserStopped) {
+      dexai::log()->info(
+          "CommInterface:HandleSimDriverEventTrigger: received "
+          "command "
+          "to simulate U Stop = True");
+      robot_data_.robot_state.robot_mode = franka::RobotMode::kUserStopped;
+    } else {
+      dexai::log()->info(
+          "CommInterface:HandleSimDriverEventTrigger: received "
+          "command "
+          "to simulate U Stop = False");
+      robot_data_.robot_state.robot_mode = franka::RobotMode::kIdle;
+    }
```

Comment:
Instead of this being a toggle, can you just check the value of `cmd_msg->data` to follow same paradigm as for pause? i.e. if it's 0 you want it to not be u-stopped, and if it's nonzero you want it to u-stop

Reply:
if (cmd_msg->data) {
// set to u-stopped
} else {
// set to not u-stopped
}
```